v0/destinations/blueshift/transform.js

```diff
@@ -0,0 +1,114 @@
+const { EventType } = require("../../../constants");
+const {
+  CustomError,
+  constructPayload,
+  ErrorMessage,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  getValueFromMessage,
+  getFieldValueFromMessage,
+  removeUndefinedAndNullValues
+} = require("../../util");
+
+const {
+  MAPPING_CONFIG,
+  CONFIG_CATEGORIES,
+  BASE_URL_EU,
+  BASE_URL
+} = require("./config");
+
+function checkValidEventName(str) {
+  if (
+    str.indexOf(".") !== -1 ||
+    str.indexOf(" ") !== -1 ||
+    /[0-9]/.test(str) ||
+    str.length > 64
+  )
+    return true;
+  return false;
+}
+
+const responseBuilderSimple = (message, category, destination) => {
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+
+  if (EventType.GROUP === message.type.toLowerCase()) {
+    const customer = removeUndefinedAndNullValues({
+      id: getFieldValueFromMessage(message, "userId"),
+      email: getFieldValueFromMessage(message, "email")
+    });
+    if (!customer.id && !customer.email) {
+      throw new CustomError(
+        "customer_id or email is required to identify customer.",
+        400
+      );
+    }
+    if (customer.id) {
+      payload.identifier_key = "customer_id";
+      payload.identifier_value = customer.id;
+    } else {
+      payload.identifier_key = "email";
+      payload.identifier_value = customer.email;
+    }
+  }
+
+  const response = defaultRequestConfig();
+  let endpoint;
+  if (destination.Config.datacenterEU) {
+    endpoint = `${BASE_URL}${category.endpoint}`;
+  } else {
+    endpoint = `${BASE_URL_EU}${category.endpoint}`;
+  }
+
+  if (EventType.GROUP === message.type.toLowerCase()) {
+    response.endpoint = endpoint.replace(":list_id", payload.list_id);
+  } else {
+    response.endpoint = endpoint;
+  }
+
+  response.method = defaultPostRequestConfig.requestMethod;
+  response.headers = {
+    "Content-Type": "application/json"
+  };
+  response.body.JSON = payload;
+  return response;
+};
+
+const processEvent = (message, destination) => {
+  if (!message.type) {
+    throw new CustomError(ErrorMessage.TypeNotSupported, 400);
```

Comment:
Lets throw descriptive error message `"Message Type is not present. Aborting message."`. This one is returning `"Message type not supported"`

Reply:
Updated the message.